lib/Parser/XML.php

```diff
@@ -0,0 +1,264 @@
+     */
+    protected $root;
+
+    /**
+     * Creates the parser.
+     *
+     * Optionally, it's possible to parse the input stream here.
+     *
+     * @param mixed $input
+     * @param int $options Any parser options (OPTION constants).
+     * @return void
+     */
+    public function __construct($input = null, $options = 0) {
+
+        if (0 === $options) {
+            $options = parent::OPTION_FORGIVING;
+        }
+
+        parent::__construct($input, $options);
+
+    }
+
+    /**
+     * Parse xCal or xCard.
+     *
+     * @param resource|string $input
+     * @param int $options
+     * @throws \Exception
+     * @return Sabre\VObject\Document
+     */
+    public function parse($input = null, $options = 0) {
+
+        if (!is_null($input)) {
+            $this->setInput($input);
+        }
+
+        if (0 !== $options) {
+            $this->options = $options;
+        }
+
+        if (is_null($this->input)) {
+            throw new EofException('End of input stream, or no input supplied');
+        }
+
+        if ($this->input['name'] === '{' . self::XCAL_NAMESPACE . '}icalendar') {
+
+            $this->root = new VCalendar([], false);
+            $this->pointer = &$this->input['value'][0];
+            $this->parseVcalendarComponents($this->root);
+
+        } else {
+            throw new \Exception('Unsupported XML standard');
+        }
+
+        return $this->root;
+    }
+
+    /**
+     * Parse a vCalendar.
+     *
+     * @param Sabre\VObject\Component $parentComponent
+     */
+    protected function parseVcalendarComponents(Component $parentComponent) {
+
+        foreach ($this->pointer['value'] ?: [] as $children) {
+
+            switch (static::getTagName($children['name'])) {
+
+                case 'properties':
+                    $xmlProperties = $children['value'];
+
+                    foreach ($xmlProperties as $xmlProperty) {
+
+                        $propertyName       = static::getTagName($xmlProperty['name']);
+                        $propertyValue      = [];
+                        $propertyParameters = [];
+                        $propertyType       = 'text';
+
+                        foreach ($xmlProperty['value'] as $i => $xmlPropertyChild) {
+
+                            if (   !is_array($xmlPropertyChild)
+                                || 'parameters' !== static::getTagName($xmlPropertyChild['name']))
+                                continue;
+
+                            $xmlParameters = $xmlPropertyChild['value'];
+
+                            foreach ($xmlParameters as $xmlParameter)
+                                $propertyParameters[static::getTagName($xmlParameter['name'])]
+                                    = $xmlParameter['value'][0]['value'];
+
+                            array_splice($xmlProperty['value'], $i, 1);
+
+                        }
+
+                        switch ($propertyName) {
+
+                            case 'dtstart':
```

Comment:
`geo`, `request-status` etc. are [special cases according to the RFC](http://tools.ietf.org/html/rfc6321#section-3.4.1).
`dstart` is a bug fix. I have to dig more about this one. See my comment here https://github.com/Hywan/sabre-vobject/commit/46ef57ceab8cb26fa89552da9a4dfb45fe1519e4#commitcomment-9279424.